uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-available-balance.component.js

```diff
@@ -78,43 +72,45 @@ define([
     }
 
     /**
-     * Loads entitlements for the selected user and absence period and maps
-     * the entitlements to their corresponding absence type.
+     * Loads entitlements for the selected user and absence period.
      *
      * @return {Promise} - Returns an empty promise when all entitlements have
      * been loaded and mapped.
      */
     function loadEntitlements () {
       return Entitlement.all({
-        contact_id: vm.contactId,
-        period_id: vm.absencePeriod.id,
-        type_id: { IN: getContractEntitlementsIds() }
+        'contact_id': vm.contactId,
+        'period_id': vm.absencePeriod.id,
+        'type_id.is_active': true
       }, true)
       .then(function (_entitlements_) {
         entitlements = _entitlements_;
-
-        mapAbsenceTypesWithTheirEntitlements();
       });
     }
 
     /**
-     * Maps absence types with their entitlements. Only absence types the user
-     * is entitled to are mapped (entitlement.value > 0). The .remainder.future
-     * is used to display the current balance for approved and open requestes.
+     * Returns an array of leave entitlements which have a value greater
+     * than zero, allows for overuse, or allows accruals of leave requests.
+     *
+     * @param {Array} absenceTypes - An array of absence types.
+     * @param {Array} entitlements - An array of entitlements.
+     * @return {Array}
      */
-    function mapAbsenceTypesWithTheirEntitlements () {
-      vm.absenceTypeEntitlements = [];
+    function getLeaveEntitlements (absenceTypes, entitlements) {
+      var entitlement;
+      var indexedEntitlements = _.indexBy(entitlements, 'type_id');
```

Comment:
too generic of a name, indexed by what? `entitlementsByAbsenceType` would be a better choice